classes/manager.php

```diff
@@ -455,4 +455,105 @@ public static function require_auth($courseorid = null, $autologinguest = null,
             }
         }
     }
+
+    /**
+     * Sets config variable for given factor.
+     *
+     * @param array $data
+     * @param string $factor
+     *
+     * @return bool true or exception
+     * @throws dml_exception
+     */
+    public static function set_factor_config($data, $factor) {
+        foreach ($data as $key => $newvalue) {
+            $oldvalue = get_config($factor, $key);
```

Comment:
Probably not your code, but I'd move this out of the loop. This will force set a cache item and then invalidte it

so grab the whole plugin config in one go at the start, then set each one which will invalidate it without rebuilding it